### PR TITLE
Add 1.4.0 migration guide.

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-931](https://youtrack.jetbrains.com/issue/KTOR-931)

### CODE REVIEW

**Comment by Stexxe** [comment]
> ```
> Fix #KTOR-931
> ```

Может лучше делать #KTOR-931 ссылкой? [KTOR-931](https://youtrack.jetbrains.com/issue/KTOR-931)

File: quickstart/migration/1.4.0.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+---
+title: 1.4.0
+caption: Migrating 1.3.2 → 1.4.0
+category: quickstart
+---
+
+## 1.4.0
+{: #1.4.0 }
```
Comment (Stexxe) [inline quickstart/migration/1.4.0.md:8]: Github rich text previewer shows this as `{: #1.4.0 }`

File: quickstart/migration/1.4.0.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+* Upgrade to kotlin 1.4.0
```
Comment (Stexxe) [inline quickstart/migration/1.4.0.md:10]: Why do we need to have a list of changes in the migration guide?

File: quickstart/migration/1.4.0.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+* Fix client.join to wait engine close
+* Verify receive from response in exception
+* Update okhttp version to 4.4.0
+* Cleanup iOS proxy configuration
+* Fix WebSocketTest.testMaxSize
+* Fix UTF8 parsing (#1718).
+
+## Migration notes
+
+* The `-native` suffix is obsolete dut to migration to HMPP. The common artifact can be used instead. For example the `kotlinx-client-core-native` -> `ktor-client-core`.
```
Comment (Stexxe) [inline quickstart/migration/1.4.0.md:77]: Typo `due`

File: quickstart/migration/1.4.0.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+* Maven users can fail to resolve `ktor-client-cio` because it's common from 1.4.0. You can use `ktor-client-cio-jvm` instead. Gradle users can use any artifact(due to metadata).
```
Comment (Stexxe) [inline quickstart/migration/1.4.0.md:78]: I think `can fail` is not appropriate here — `may fail` or `will fail` is more suitable, cc @hhariri.

Comment (Stexxe) [reply quickstart/migration/1.4.0.md:78]: Space before `(due to metadata)` is missing.

**Comment by e5l** [comment]
Sure, done

File: quickstart/migration/1.4.0.md (hunk 3 above)
Comment (Stexxe) [inline quickstart/migration/1.4.0.md:77]: This paragraph is difficult to read because it contains many small sentences.

**Review by Stexxe — CHANGES_REQUESTED** [review]

File: quickstart/migration/1.4.0.md (hunk 2 above)
Comment (e5l) [reply quickstart/migration/1.4.0.md:10]: We had them for the previous: https://ktor.io/quickstart/migration/1.3.0.html
@hhariri what do you think?

File: quickstart/migration/1.4.0.md (hunk 1 above)
Comment (e5l) [reply quickstart/migration/1.4.0.md:8]: It's a jekyll syntax

File: quickstart/migration/1.4.0.md (hunk 3 above)
Comment (e5l) [reply quickstart/migration/1.4.0.md:77]: Thanks

File: quickstart/migration/1.4.0.md (hunk 2 above)
Comment (Stexxe) [reply quickstart/migration/1.4.0.md:10]: Maybe we should make this section hideable/scrollable or less perceptible than the guide.

File: quickstart/migration/1.4.0.md (hunk 3 above)
Comment (e5l) [reply quickstart/migration/1.4.0.md]: Could you tell me if you have an idea of rewriting it?

Comment (Stexxe) [reply quickstart/migration/1.4.0.md]: I suggest asking @hhariri how to better structure it.

File: quickstart/migration/1.4.0.md (hunk 2 above)
Comment (hhariri) [reply quickstart/migration/1.4.0.md:10]: We're eventually going to move this to What's New on the site. Not sure it's needed in Migration Guide.

File: quickstart/migration/1.4.0.md (hunk 3 above)
Comment (hhariri) [reply quickstart/migration/1.4.0.md]: The `-native` suffix is now obsolete as per the new Kotlin Multiplatform Project approach, which allows the use of common artifacts. This means that for instance `kotlinx-client-core-native` would be replaced with `ktor-client-core`.

Comment (e5l) [reply quickstart/migration/1.4.0.md]: Thanks

**Pull request merged**